File: internal/domain/friend.go
package domain

import "time"

type Friendship struct {
    ID          int64     `json:"id"`
    RequesterID int64     `json:"requester_id"`
    AddresseeID int64     `json:"addressee_id"`
    Status      string    `json:"status"` // pending, accepted, rejected, cancelled
    CreatedAt   time.Time `json:"created_at"`
    UpdatedAt   time.Time `json:"updated_at"`
}

// Публичная карточка пользователя для поиска и списка друзей
type UserPublicCard struct {
	ID           int64   `json:"id"`
	FriendshipID *int64  `json:"friendship_id,omitempty"`
	Nickname     *string `json:"nickname,omitempty"`
	Username     *string `json:"username,omitempty"`
	FullName     string  `json:"full_name"`
	AvatarURL    *string `json:"avatar_url,omitempty"`
	University   *string `json:"university,omitempty"`
}

// Статус дружбы между текущим пользователем и другим
type FriendStatusResponse struct {
    UserID  int64  `json:"user_id"`
    Status  string `json:"status"` // none, pending_outgoing, pending_incoming, friends
    FriendshipID *int64 `json:"friendship_id,omitempty"`
}

File: ios/StudentDiscountApp/Sources/Models/Friend.swift
import Foundation

struct UserPublicCard: Codable, Identifiable {
    let id: Int64
    let friendshipID: Int64?
    let nickname: String?
    let username: String?
    let fullName: String
    let avatarURL: String?
    let university: String?

    enum CodingKeys: String, CodingKey {
        case id, nickname, username, university
        case friendshipID = "friendship_id"
        case fullName = "full_name"
        case avatarURL = "avatar_url"
    }

    var displayName: String {
        nickname ?? fullName
    }
}

struct Friendship: Codable, Identifiable {
    let id: Int64
    let requesterID: Int64
    let addresseeID: Int64
    let status: String

    enum CodingKeys: String, CodingKey {
        case id, status
        case requesterID = "requester_id"
        case addresseeID = "addressee_id"
    }
}
